alonzo/formal-spec/utxo.tex

```diff
@@ -54,16 +51,14 @@ \section{UTxO}
   \begin{enumerate}[label=({\roman*})]
     \item the fee amount that the transaction states it is paying suffices to cover
     the minimum fee that the transaction is obligated to pay; and if the transaction uses non-native scripts, that
-    \item the fee-marked inputs do not belong to non-native script addresses (that is, they
-    belong to either a native script address, or a key hash one);
-    \item all the fee-marked inputs contain strictly Ada and no other kinds of token; and
-    \item the fee-marked inputs are sufficient to cover the fee amount that is stated
-    in the transaction.
+    \item the collateral inputs are strictly in key-hash addresses;
+    \item all the collateral inputs contain strictly Ada and no other kinds of token; and
```

Comment:
I think we can relax this restriction for simpler UTxO selection, and since tx is deterministic, there should be no problem, the one who send invalid transaction should fully acknowledge the consequence(burn of collaterals).

Reply:
This is true - but so far, non-ada assets can only be burned if the minting/burning policy is satisfied. There are plans to change that (eg. create a address to send tokens you are burning to), but I it would be weird if before that is possible, the only way to burn tokens without a valid policy is to have a failing transaction... So I think this relaxation change would not be good to make now

Reply:
I see, thanks.

Reply:
> but I it would be weird if before that is possible, the only way to burn tokens without a valid policy is to have a failing transaction

Normally, the incentive to burn token is to reclaim the minimal ada, but burn collaterals don't get you anything back, so I don't think it'll become a mainstream way to do burning.